### PR TITLE
Add some helpers and patch ups to the integration tests

### PR DESCRIPTION
**Pull request by nathanleclaire**
cc @ehazlett @hairyhenderson 

This PR:

1. Cleans up some existing integration test issues
2. Introduces some new helpers to enforce test conditions
3. Adds experimental tests for RancherOS and RedHat
4. Displays test name in the wrapper script output - it was a kind of hard to follow before

Signed-off-by: Nathan LeClaire <nathan.leclaire@gmail.com>

### CODE REVIEW

File: test/integration/provision/os/rancheros.bats [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+load ${BASE_TEST_DIR}/helpers.bash
+
+force_env DRIVER virtualbox
+
+export RANCHEROS_VERSION="v0.3.1"
+export RANCHEROS_ISO="https://github.com/rancherio/os/releases/download/$RANCHEROS_VERSION/machine-rancheros.iso"
+
+@test "$DRIVER: create with RancherOS ISO" {
+  VIRTUALBOX_BOOT2DOCKER_URL="$RANCHEROS_ISO" run ${BASE_TEST_DIR}/run-bats.sh ${BASE_TEST_DIR}/core
```
Comment (nathanleclaire) [inline test/integration/provision/os/rancheros.bats]: This test actually doesn't work with _all_ of the tests in `core` due to a variety of issues (e.g. engine options test expects `--engine-storage-driver devicemapper` to be allowed with RancherOS, which it isn't), so I will probably change it around a bit, but I wanted to get it out the door as a POC

**Comment by ehazlett** [comment]
LGTM

**Comment by nathanleclaire** [comment]
Ah, before you merge this, that removed check for `DRIVER` should actually be put back in.  I was experimenting with `force_env` setting DRIVER automatically, but I would prefer to always explicitly set it, otherwise it's less safe.  Will update shortly

**Comment by nathanleclaire** [comment]
OK, I've fixed the removed check.  This is good to merge now.

**Comment by ehazlett** [comment]
@hairyhenderson PTAL

**Comment by hairyhenderson** [comment]
Sorry, thought I already commented on this but clearly I didn't... LGTM :+1:

**Comment by ehazlett** [comment]
@hairyhenderson thanks!
